File: 프로그래머스/프로그래머스_Lv2/숫자 카드 나누기/solution.js
function dividing(arr, num) {
  for (let n of arr) {
    if (n % num == 0) continue;
    else return false;
  }

  return true;
}

function notDividing(arr, num) {
  for (let n of arr) {
    if (n % num != 0) continue;
    else return false;
  }

  return true;
}

function solution(arrayA, arrayB) {
  let answer = 0;
  const num = Math.max(arrayA[0], arrayB[0]);
  for (let i = 2; i <= num; i++) {
    if (
      (dividing(arrayA, i) && notDividing(arrayB, i)) ||
      (dividing(arrayB, i) && notDividing(arrayA, i))
    ) {
      answer = i;
    }
  }
  return answer;
}

File: 프로그래머스/프로그래머스_Lv2/숫자 카드 나누기/Solution.java
import java.util.*;
import java.lang.Math;
class Solution {
    public boolean dividing(int[] arr, int num){
        for(int n : arr){
            if(n%num == 0) continue;
            else return false;
        }

        return true;
    }

    public boolean notDividing(int[] arr, int num){
        for(int n : arr){
            if(n%num != 0) continue;
            else return false;
        }

        return true;
    }

    public int solution(int[] arrayA, int[] arrayB) {
        int answer = 0;
        int num = Math.max(arrayA[0], arrayB[0]);
        for(int i=2; i<=num; i++){
            if((dividing(arrayA, i) && notDividing(arrayB,i)) || 
              (dividing(arrayB, i) && notDividing(arrayA,i))){
                answer = i;
            }
        }
        return answer;
    }
}